compiler/rustc_hir_analysis/src/check/wfcheck.rs

```diff
@@ -307,7 +307,11 @@ fn check_item<'tcx>(tcx: TyCtxt<'tcx>, item: &'tcx hir::Item<'tcx>) -> Result<()
             check_item_type(tcx, def_id, ty.span, UnsizedHandling::Forbid)
         }
         hir::ItemKind::Const(ty, ..) => {
-            check_item_type(tcx, def_id, ty.span, UnsizedHandling::Forbid)
+            let res = check_item_type(tcx, def_id, ty.span, UnsizedHandling::Forbid);
+            res.and(enter_wf_checking_ctxt(tcx, item.span, def_id, |wfcx| {
```

Comment:
fast path if there is no w.c.

Reply:
Ah, I didn't actually intend to enter the wfchecking ctxt twice, once in `check_item_type` and once here